### PR TITLE
ci: A few fixes of `ccache` issues

### PR DESCRIPTION
**Pull request by hebasto**
Combined https://github.com/bitcoin/bitcoin/pull/27077 and https://github.com/bitcoin/bitcoin/pull/27083 for the sake of more representative testing.

The [second](https://cirrus-ci.com/build/5852404411793408) run looks promising:
- ["MSan"](https://cirrus-ci.com/task/4583485717872640) task -- 98.74 %
- ["ASan + LSan + UBSan + integer"](https://cirrus-ci.com/task/5146435671293952) task -- 100.0 %
- ["multiprocess"](https://cirrus-ci.com/task/4864960694583296) task -- 99.35 %
- ["macOS 13 native"](https://cirrus-ci.com/task/6342704322314240) task -- 100.0%
- ["Android APK"](https://cirrus-ci.com/task/6272335578136576) task -- 100.0%

Although, the ["previous releases"](https://cirrus-ci.com/task/4935329438760960) task has a 66.10 % cache hit rate. It needs more investigation (not in this PR).

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -5,8 +5,9 @@ env:  # Global defaults
   TEST_RUNNER_PORT_MIN: "14000"  # Must be larger than 12321, which is used for the http cache. See https://cirrus-ci.org/guide/writing-tasks/#http-cache
   CI_FAILFAST_TEST_LEAVE_DANGLING: "1"  # Cirrus CI does not care about dangling process and setting this variable avoids killing the CI script itself on error
   CCACHE_SIZE: "200M"
-  CCACHE_DIR: "/tmp/ccache_dir"
+  CCACHE_DIR: '${CIRRUS_WORKING_DIR}/ccache_dir'
```
Comment (MarcoFalke) [inline .cirrus.yml:8]: This will still be in the same volume (tmp), so why is this changed?

Comment (hebasto) [reply .cirrus.yml:8]: I think the difference is not using absolute path `/tmp/...`.

**Comment by hebasto** [comment]
The PR description has been updated with some statistics.

File: .cirrus.yml (hunk 1 above)
Comment (MarcoFalke) [reply .cirrus.yml:8]: I am still not sure what this means and which paths are preferred or discouraged

**Comment by fanquake** [comment]
Most of the changes here look straight forward enough, and clearly improve `ccache` usage, but this is blocked on at least an answer to this question: https://github.com/bitcoin/bitcoin/pull/27084#discussion_r1104757809.

**Comment by hebasto** [comment]
@fanquake @MarcoFalke 

> ... but this is blocked on at least an answer to this question: [#27084 (comment)](https://github.com/bitcoin/bitcoin/pull/27084#discussion_r1104757809).

I've dropped the first commit and faced some weird issues which I saw in #27083 before. Going to provide steps to reproduce that unwanted behavior.

Making this PR a draft for now.

File: .cirrus.yml (hunk 1 above)
Comment (hebasto) [reply .cirrus.yml:8]: https://github.com/bitcoin/bitcoin/pull/27084#issuecomment-1438041890

**Comment by hebasto** [comment]
The questionable commit has been dropped.

Ready to re-review.

**Review by maflcko — APPROVED** [review]
lgtm

File: .cirrus.yml [hunk 2]
```diff
@@ -36,6 +36,8 @@ main_template: &MAIN_TEMPLATE
   timeout_in: 120m  # https://cirrus-ci.org/faq/#instance-timed-out
   ccache_cache:
     folder: "/tmp/ccache_dir"
+    fingerprint_script: echo ${CIRRUS_TASK_NAME}
```
[the remark this answers is not in the document]
Comment (fanquake) [reply .cirrus.yml:40]: > Isn't this already the default?

Yep looks like it.

**Comment by hebasto** [comment]
> I guess instead of `${CIRRUS_WORKING_DIR}/ccache_dir` you can also just write `ccache_dir`, as `CIRRUS_WORKING_DIR` is implicit?

While it works for the `folder` property  of the `ccache_cache`, it seems the `CCACHE_DIR` variable is supposed to contain an absolute path.

**Comment by fanquake** [comment]
What is the status of this

**Comment by hebasto** [comment]
> What is the status of this

I've analyzed the [recent](https://cirrus-ci.com/build/5086582458286080) `ccache` summaries at https://github.com/bitcoin/bitcoin/commit/a75c77ea903c100531e0fc5fde94bb9b52642145.

I don't think this PR can improve them.
